src/main/java/com/bloxbean/cardano/client/address/util/AddressEncoderDecoderUtil.java

```diff
@@ -52,6 +54,10 @@ public static NetworkId getNetworkId(Network networkInfo) {
             network = NetworkId.MAINNET;
```

Comment:
I think we need to change this logic. In existing code, Network object is being compared but instead we should compare NetworkId of the network and return NetworkId.MAINNET or NetworkId.TESTNET


Some thing like
```
if (NetworkId.MAINNET.equals(networkInfo.getNetworkId()) return NetworkId.MAINNET;
else  if //check for network TESTNET network id

else throw AddressRuntimeException
```